conda/common/configuration.py

```diff
@@ -1600,3 +1601,56 @@ def typify_parameter(self, parameter_name, value, source):
 
     def get_descriptions(self):
         raise NotImplementedError()
+
+
+def unique_sequence_map(
+    *, unique_key: str, property_name: str, allowed_keys: set | None = None
```

Comment:
I don't think I agree with the `allowed_keys` as implemented here. This means our supported reporters are hardcoded in conda, instead I envision reporter plugins. Something more akin to what we do for the solver.

Reply:
@kenodegard,

One way to solve this is by defining this function argument as a callback function which will allow it to be lazy-loaded at runtime.

Does this sound like a good idea to you?

Reply:
But, I will also state here that I find it very unlikely that "reporters" will be turned into a plugin in the future, so I believe this could be over optimization for a problem that is unlikely to ever arise.

Reply:
Why can't these be plugin hooks from the get-go? Plugins aren't just for third-party use to add new functionality but also as a way for us to modularize and organize stuff within conda.

There are absolutely cases where additional reporters could be defined, e.g., conda-tui could define a new type of reporter that uses textual instead of rich or Navigator could desire a custom reporter to interface with conda instead of relying on the builtin json reporter.

I think about our plugin initiatives through the lens of "if we already had this plugin hook what could we have already accomplished sooner/faster?" And in this case, if we had a reporter hook previously defined, a third-party plugin could easily have introduced rich support years ago with minimal effort.

Reply:
I hear those concerns, but I believe that discussion is better to be had in a separate issue. I would really like to keep this implementation as simple as possible for now and not have to worry about the complexities surrounding setting up another plugin hook for this.

In the future, this should be pretty easy to retro-fit to support that use case. Like I said, we could simply change this decorator function to optionally accept a callback function to determine all possible values for `allowed_keys`. That will allow us to dynamically determine the value of that argument at runtime and therefore take into account all the possible values registered by plugin hooks.

So, in order to keep this pull request going forward, I kindly ask that you open a separate issue and attach it to the ongoing epic for this work:

- https://github.com/conda/conda/issues/13707

I would be more than happy to have a conversation about plugin hooks there.